core/arch/arm/kernel/boot.c

```diff
@@ -1241,6 +1282,14 @@ static void init_secondary_helper(unsigned long nsec_entry)
 	init_vfp_sec();
 	init_vfp_nsec();
 
+#ifdef CFG_CORE_HALT_CORES_ON_PANIC
+	if (interrupt_add_handler_with_chip(itr_chip,
+					    &multi_core_panic_handler))
+		panic();
+
+	interrupt_enable(itr_chip, multi_core_panic_handler.it);
```

Comment:
Should be called only once in case there are more than 2 cores.

Reply:
Wouldn't it make sense to put this in a boot_final() function instead? We could even move the code to `core/kernel/panic.c` if we could only resolve the masking of timer interrupts.

Reply:
+1. Having the implementation in _panic.c_ fully makes sense to me. `init_secondary_helper()` would call the feature initialization.

Reply:
> Should be called only once in case there are more than 2 cores.

Would adding a static assert: static_assert(CFG_TEE_CORE_NB_CORE > 1) be preferable?

-----------------------------------
> Wouldn't it make sense to put this in a boot_final() function instead? We could even move the code to `core/kernel/panic.c` if we could only resolve the masking of timer interrupts.

I've done it this way so that it's called when the secondary CPU init is done.

I can move the handler init code to _panic.c_  and call it here in init_secondary_helper() if IS_ENABLED(CFG_CORE_HALT_CORES_ON_PANIC) if that's ok?

Reply:
> > Should be called only once in case there are more than 2 cores.
> 
> Would adding a static assert: static_assert(CFG_TEE_CORE_NB_CORE > 1) be preferable?

No, I mean if there are 4 core, the 3 secondary cores will go through this but while each secondary core needs to enable/unmask its SGI interrupt (`interrupt_enable()`) only 1 core shall to register the interrupt callback handler (`interrupt_add_handler_with_chip()`) unless what the handler will be registered several times which is not what you want and will even fail since `ITRF_SHARED` is not set.

> I can move the handler init code to panic.c and call it here in init_secondary_helper() if IS_ENABLED(CFG_CORE_HALT_CORES_ON_PANIC) if that's ok?

+1

Reply:
Sorry I misread your comment. I'll add a static variable to control that